### PR TITLE
[Requirements] Drop Python2 Support

### LINKED ISSUE #82: [requirements] Update dependencies & drop python 2 support
Python2 is very close to the end of it's lifecycle, it's time we drop the support for Python 2.

To do that,
1. We need to update all the dependencies to python3.
2. Remove code that exists for python 2 support.
3. Remove python2 from README & any documentation.

### PR DESCRIPTION
**Pull request by IAmAMissingColon**
* Upgraded dependencies
* Removed structures supported by `python2`
* simplified `super()` function
* Upgraded to Django3

Closes #82

### CODE REVIEW

**Comment by IAmAMissingColon** [comment]
* As I think the `django1` should be remove as it's not supporting `django-allauth` anymore
* Should I remove the python3.5 as well ?

**Comment by IAmAMissingColon** [comment]
> @IAmAMissingColon I tested with [openwisp/openwisp-utils#44](https://github.com/openwisp/openwisp-utils/pull/44) & `django-model-utils==4.0.0` but the tests are failing, can you please investigate into it?

Okay :smile:

**Comment by IAmAMissingColon** [comment]
> About your questions,
> 
> 1. Yes, if it's required due to dependency, please drop it.
> 2. I think yes we can drop python 3.5, it's about to be deprecated in 9 months https://devguide.python.org/#status-of-python-branches

Okay! :smile: 

Dropping Python3.5

**Comment by IAmAMissingColon** [comment]
> @IAmAMissingColon I tested with [openwisp/openwisp-utils#44](https://github.com/openwisp/openwisp-utils/pull/44) & `django-model-utils==4.0.0` but the tests are failing, can you please investigate into it?

It shows `openwisp-utils 0.3 has requirement django-model-utils<3.3.0,>=3.1.2, but you'll have django-model-utils 4.0.0 which is incompatible.` maybe that's the reason of fail

**Comment by IAmAMissingColon** [comment]
The tests for Django3 are passing but somehow the Django2.1 is failing! @atb00ker

**Comment by IAmAMissingColon** [comment]
@atb00ker It might be realted to Django2.1?

File: .travis.yml [hunk 1]
```diff
@@ -2,23 +2,13 @@ language: python
...
-
-matrix:
-  exclude:
-   - python: "2.7"
-     env: DJANGO="django>=2.0,<2.1"
-   - python: "2.7"
-     env: DJANGO="django>=2.1,<2.2"
-   - python: "2.7"
-     env: DJANGO="django>=2.2,<2.3"
+  - DJANGO="django>=3.0,<3.2"
```
Comment (atb00ker) [inline .travis.yml]: "<3.2" ? Did you just make this change or what this before as well?
This should clearly be 3.1, why is this 3.2?

Comment (IAmAMissingColon) [reply .travis.yml]: Typo :sweat_smile:

**Comment by IAmAMissingColon** [comment]
Just that? Nothing else is remaining ?? :smile:

**Comment by IAmAMissingColon** [comment]
Rebased with master :smile: , but Django3 tests are failing now.. fixing them :smile:

File: requirements-test.txt [hunk 2]
```diff
@@ -1,5 +1,3 @@
 coveralls
-isort
-flake8
-openwisp-utils[qa]>=0.2.1
+openwisp-utils[qa]>=0.3.2
```
Comment (nemesisdesign) [inline requirements-test.txt]: set this to `0.4.0`

Comment (IAmAMissingColon) [reply requirements-test.txt]: Done, but still it's showing the same conflict :expressionless:

**Comment by IAmAMissingColon** [comment]
@nemesisdesign Done with your changes :smile: , but is there any way I can increase coverage ?

**Pull request merged**